embark.el

```diff
@@ -526,18 +526,22 @@ In `dired-mode', it uses `dired-get-filename' instead."
 
 (defun embark-target-expression-at-point ()
   "Target expression at point."
-  (pcase (bounds-of-thing-at-point 'sexp)
-    ((and bounds `(,begin . ,end))
-     (let ((pt (point)))
-       (when (or (and (= pt begin)
-                      (memq (syntax-class (syntax-after pt)) '(4 6 7)))
-                 (and (= pt end)
-                      (memq (syntax-class (syntax-after (1- pt))) '(5 7)))
-                 (and (= pt (1+ begin))
-                      (eq (syntax-class (syntax-after begin)) 6)
-                      (eq (syntax-class (syntax-after pt)) 4)
-                      (setq begin (1+ begin))))
-         `(expression ,(buffer-substring begin end) . ,bounds))))))
+  (save-excursion
+    (let ((pt (point)))
+      (catch 'found
+        (while
+            ;; Looking at opening parenthesis or find last one
+            (or (memq (syntax-class (syntax-after pt)) '(4 6 7))
+                (re-search-backward "\\(\\s(\\|\\s/\\|\\s\"\\)" nil 'noerror))
```

Comment:
I would have thought that using (bounds-of-thing-at-point 'list) would have been easier. Was there some problem with that approach?

Reply:
Oh, I wasn't aware of 'list, does it do the same expansion as I am doing here?

Reply:
Okay, this seems to work as expected. Minor difference is that the point must be really inside the sexp, while my finder also works if you are on the parentheses. But this is probably not worth the complication!

~~~ elisp
(defun embark-target-expression-at-point ()
  "Target expression at point."
  (when-let (bounds (bounds-of-thing-at-point 'list))
    `(expression ,(buffer-substring (car bounds) (cdr bounds)) . ,bounds)))
~~~

Reply:
Oh, and it doesn't work with strings. Hmm, I am not sure. Maybe keep my more complicated version?

Reply:
Yes. Maybe in the future I'll add a separate string target, just to include the fill-paragraph action for it. When that happens we could maybe simplify this one (but keeping the matching on the parentheses).